contrib/src/main/java/com/datatorrent/contrib/parser/AbstractCsvParser.java

```
@@ -0,0 +1,430 @@
+        }
+      }
+      catch (IOException ex) {
+        DTThrowable.rethrow(ex);
+      }
+
+    }
+
+      int countKeyValue = fields.size();
+      properties = new String[countKeyValue];
+      processors = new CellProcessor[countKeyValue];
+      initialise(properties, processors);
+      CsvPreference preference = new CsvPreference.Builder('"', fieldDelimiter, lineDelimiter).build();
+      csvReader = getReader(csvStringReader, preference);
+
+  }
+
+  // Initialise the properties and processors.
+  public void initialise(String[] properties, CellProcessor[] processors)
+  {
+    for (int i = 0; i < fields.size(); i++) {
+      FIELD_TYPE type = fields.get(i).type;
+      properties[i] = fields.get(i).name;
+      if (type == FIELD_TYPE.DOUBLE) {
+        processors[i] = new Optional(new ParseDouble());
+      }
+      else if (type == FIELD_TYPE.INTEGER) {
+        processors[i] = new Optional(new ParseInt());
+      }
+      else if (type == FIELD_TYPE.FLOAT) {
+        processors[i] = new Optional(new ParseDouble());
+      }
+      else if (type == FIELD_TYPE.LONG) {
+        processors[i] = new Optional(new ParseLong());
+      }
+      else if (type == FIELD_TYPE.SHORT) {
+        processors[i] = new Optional(new ParseInt());
+      }
+      else if (type == FIELD_TYPE.STRING) {
+        processors[i] = new Optional();
+      }
+      else if (type == FIELD_TYPE.CHARACTER) {
+        processors[i] = new Optional(new ParseChar());
+      }
+      else if (type == FIELD_TYPE.BOOLEAN) {
+        processors[i] = new Optional(new ParseChar());
+      }
+      else if (type == FIELD_TYPE.DATE) {
+        processors[i] = new Optional(new ParseDate("dd/MM/yyyy"));
+      }
+    }
+
+  }
+
+  @Override
+  public void teardown()
+  {
+    try {
+      csvReader.close();
+    }
+    catch (IOException e) {
+      logger.error("Parsing csv error", e);
```

Don't swallow exception.